collections.md

```diff
@@ -444,6 +449,124 @@ Return codes:
 * 403 Forbidden - if you are not logged in with sufficient permissions to retrieve the current collection or parent community
 * 404 Not found - if the current collection doesn't exist
 
+### Groups
+
+This includes group management for the type of groups:
+* groups/admin: the Collection Administrator group
+* groups/submit: the Collection Submitters group
+* groups/itemread: the Collection Default item READ rights group
+* groups/bitstreamread: the Collection Default bitstream READ rights group
+* groups/workflow/<:workflow-role>: the Collection Workflow groups
+
+#### Collection administrators
+**/api/core/collections/<:uuid>/groups/admin**
```

Comment:
While I understand the desire here to create two sub resources to "subdivide" types of groups, this `/groups/admin` path seems a bit odd to me.  For one, it implies that `/groups` should return a list of available Collection groups & that you could potentially add more "types" of groups to a Collection.

I know it's not as "pretty", but a part of me really wonders if we should just simplify this to something like `/admingroup` or even `/administrators` (as that defines the relationship that this group has with a Collection).  That said, I can see how that might not work as well for Collection Workflow Groups which are configurable.

This comment is not really a request for change -- I'm still thinking on this myself.  Just noting that the endpoints suggested here seem a little odd to me (but I'm not yet convinced either way).

Reply:
@tdonohue it is indeed not necessary for the static groups, but just to make it more "pretty" or "readable"
For the workflow groups, it can also be `/collections/<:uuid>/workflowgroups/<:workflow-role>` similar to https://github.com/DSpace/Rest7Contract/blob/master/authorities.md#authority-entry-values

So moving it to /admingroup /itemreadgroup /bitstreamreadgroup, … is not a technical problem, it just is about how "pretty" it is

Reply:
@benbosman : I think my preference here is to move more towards something like either `/adminGroup` (camel-cased) or just `/administrators` (and note in the contract that `/administrators` only accepts/returns a single Group, similar to how `collection.getAdministrators()` returns a Group)

The reason for this preference is that I think that aligns better with our REST Contract definition of what a sub-path "means".  As we note in the main README [here](https://github.com/DSpace/Rest7Contract#on-sub-path-of-a-single-resource-endpoint-associations)

> This type of endpoint interacts with (one or more) resources that are associated with a single resource. For that reason, it is sometimes called an "association" endpoint. 

Calling this subpath `/groups/admin` seems odd to me because that doesn't seem to define the association properly....as it'd be like saying "This Collection has a "group" relationship to a bunch of Groups, and one of those Groups has a unique ID of `admin`".  So, while `/groups/admin` makes a nice, human-understandable URL (for a website or similar), it's a bit odd when you look at it as an REST association.

I think calling this `/adminGroup` or `/administrators` aligns more with our definition of what a subpath means.  So, for example, in the existing endpoint `/items/<:uuid>/mappedCollections`, the subpath "mappedCollections" defines a relationship between a Collection and an Item that is *different* than say "owningCollection".  We didn't structure these endpoints like `/collections/mapped` and `/collections/owning`.    

Similarly, in this endpoint, the relationship between the `COLLECTION_*_ADMIN` Group and this Collection might be better named `/adminGroup` or `/administrators` as it clearly defines *why* this particular group is related to the Collection.

With regards to the `/workflowGroups`, your idea seems fine.   I'm OK with the endpoint  `/workflowGroups/<:workflow-role>`, as several groups have the "workflowGroup" relationship/association with this Collection...and if you need to select *one*, we need to use the `workflow-role` to do so.

Reply:
I have updated the contract based on this input
Thanks for the review